extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -91,4 +98,35 @@ public void build(CombinedIndexBuildItem indexBuildItem, BuildProducer<Reflectiv
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, StickyAssignor.class.getName()));
 
     }
+
+    /**
+     * Generate a class which replaces the usage of {@code MethodHandle} in {@code Java9ChecksumFactory} with a plain
+     * constructor invocation when run under GraalVM. This is necessary because the native image generator does not
+     * support method handles.
+     *
+     * @return the generated class
+     */
+    @BuildStep
+    public GeneratedClassBuildItem replaceJava9Code() {
+        // make our own class output to ensure that our step is run.
+        byte[][] holder = new byte[1][];
+        ClassOutput classOutput = new ClassOutput() {
+            public void write(final String name, final byte[] data) {
+                holder[0] = data;
+            }
+        };
+        try (ClassCreator cc = ClassCreator.builder().className(TARGET_JAVA_9_CHECKSUM_FACTORY)
+                .classOutput(classOutput).setFinal(true).superClass(Object.class).build()) {
+
+            cc.addAnnotation("com/oracle/svm/core/annotate/TargetClass").addValue("className",
+                    "org.apache.kafka.common.utils.Crc32C$Java9ChecksumFactory");
+            cc.addAnnotation("com/oracle/svm/core/annotate/Substitute");
+
+            try (MethodCreator mc = cc.getMethodCreator("create", Checksum.class)) {
+                mc.addAnnotation("com/oracle/svm/core/annotate/Substitute");
+                mc.returnValue(mc.newInstance(MethodDescriptor.ofConstructor("java.util.zip.CRC32C")));
+            }
+        }
+        return new GeneratedClassBuildItem(false, TARGET_JAVA_9_CHECKSUM_FACTORY, holder[0]);
```

Comment:
I might be missing the obvious but I don't understand why you need to generate the class?

Reply:
Because we build with Java 8, so we can't directly reference the class.  It was either this or switching from `MethodHandle` to reflection; this seemed the better choice.

Once we switch to Java 11 as a baseline, we can replace this with a trivial substitution class in the runtime module instead.

Reply:
This wouldn't be necessary at all if the upstream project just built an MR-jar.

Reply:
Even with a `TargetClass#onlyWith` checking that the class exists?

Reply:
It doesn't matter what annotations I use; if I try to reference `java.util.zip.CRC32C` and then build with Java 8, it's a compile error.  The class wasn't added until Java 9.

Reply:
Ah OK, sorry, I thought it was the substituted class that was a problem.